server/graphql/v2/query/collection/TransactionsCollectionQuery.ts

```diff
@@ -275,6 +280,21 @@ export const TransactionsCollectionResolver = async (args, req: express.Request)
   if (args.kind || args.kinds) {
     where.push({ kind: args.kind || args.kinds });
   }
+  if (args.paymentMethodType) {
+    const uniquePaymentMethodTypes: string[] = uniq(args.paymentMethodType.filter(Boolean));
+    const paymentMethodConditions: Record<string, string | null>[] = uniquePaymentMethodTypes.map(type => ({
+      '$PaymentMethod.type$': type,
+    }));
+
+    if (args.paymentMethodType.includes(null)) {
+      paymentMethodConditions.push({ PaymentMethodId: null });
+    }
+
+    if (paymentMethodConditions.length) {
+      include.push({ model: models.PaymentMethod });
+      where.push({ [Op.or]: paymentMethodConditions });
+    }
+  }
```

Comment:
Maybe I should add a comment here for clarity. Basically, we'll allowing to filter by `null` to get transactions without payment methods